bin/run-concurrency-test

```diff
@@ -28,7 +28,7 @@ TEST_CMD="${TEST_CMD:-"${VECTOR} -vv --config ${DIR}/concurrency/vector.toml"}"
 HTTP_TEST_SERVER="${HTTP_TEST_SERVER:-${DIR}/../http_test_server}"
 OUTPUT_DIR="${OUTPUT_DIR:-$(mktemp -d -t vector-XXXXXXXXXX)}"
 TEST_TIME=${TEST_TIME:-60} # how many seconds to run test for
-HTTP_TEST_ADDRESS=${HTTP_TEST_ADDRESS:-localhost:8080}
+HTTP_TEST_ADDRESS=localhost:0
```

Comment:
👍 to making the port dynamic. Maybe we could leave the ability to override it though? I think `bin/run-concurrency-test-suite` actually depends on that behavior though you could also modify that one to no longer require it given the changes you have here to pick an open port dynamically.

Reply:
Hmm, the test suite can then just drop that attempt to set the port, since each run will pick one at random.

Reply:
👍 could we still do this?

Reply:
Missed that this was done 😅